test/Microsoft.ML.Tests/TextLoaderTests.cs

```
@@ -7,21 +7,147 @@
+
+                using (var cursor = data.GetRowCursor((a => true)))
+                {
+                    var col1 = cursor.GetGetter<sbyte>(0);
+                    var col2 = cursor.GetGetter<short>(1);
+                    var col3 = cursor.GetGetter<int>(2);
+                    var col4 = cursor.GetGetter<long>(3);
+
+                    Assert.True(cursor.MoveNext());
+
+                    sbyte[] sByteTargets = new sbyte[] { sbyte.MinValue, sbyte.MaxValue, default};
+                    short[] shortTargets = new short[] { short.MinValue, short.MaxValue, default };
+                    int[] intTargets = new int[] { int.MinValue, int.MaxValue, default };
+                    long[] longTargets = new long[] { long.MinValue, long.MaxValue, default };
+
+                    int i = 0;
+                    for (; i < sByteTargets.Length; i++)
+                    {
+                        sbyte sbyteValue = -1;
+                        col1(ref sbyteValue);
+                        Assert.Equal(sByteTargets[i], sbyteValue);
+
+                        short shortValue = -1;
+                        col2(ref shortValue);
+                        Assert.Equal(shortTargets[i], shortValue);
+
+                        int intValue = -1;
+                        col3(ref intValue);
+                        Assert.Equal(intTargets[i], intValue);
+
+                        long longValue = -1;
+                        col4(ref longValue);
+                        Assert.Equal(longTargets[i], longValue);
+
+                        if (i < sByteTargets.Length - 1)
+                            Assert.True(cursor.MoveNext());
+                        else
+                            Assert.False(cursor.MoveNext());
+                    }
+
+                    Assert.Equal(i, sByteTargets.Length);
+                }
+            }
+            catch
+            {
+                Assert.True(false, "Test failed.");
+            }
```

If an exception gets thrown in the test, that will fail the test. But we'll also get to see what the exception is. By concealing it in this `catch` we hide the exception, which makes the test needlessly complex (since this isn't a test where we expect exceptions) and makes the result harder to interpret.